app/jobs/discord/handle_interaction_job.rb

```diff
@@ -99,9 +99,9 @@ def ping_component
     end
 
     def attach_receipt_component
-      hcb_code = HcbCode.find_by_hashid!(@params)
       discord_message = Discord::Message.find_by(discord_message_id: @interaction.dig(:message, :id))
       activity = PublicActivity::Activity.find_by(id: discord_message&.activity_id)
+      hcb_code = activity.trackable&.canonical_pending_transaction&.local_hcb_code
```

Comment:
Based on how this is written, it appears `discord_message` and `activity` may be `nil`. Should we explicitly handle that case?

At the moment, it may error when calling `trackable` on `nil`.